quakemigrate/signal/local_mag/amplitude.py

```diff
@@ -568,51 +583,48 @@ def _get_amplitude_windows(self, station, i, event, p_ttimes, s_ttimes,
 
         return windows, picked
 
-    def _get_picks(self, i, event):
+    def _get_picks(self, station, event):
         """
-        Get picks from this station for this event. Phase picks are preferred,
-        but modelled times according to the event location and travel-time
-        look-up table are used if no automatic pick was made.
+        Get picks from this station for this event. If no phase picks are
+        found, or if no pick was found, -1 is returned.
 
         Parameters
         ----------
-        i : int
-            Iterator variable.
+        station : str
+            Station name.
         event : :class:`~quakemigrate.io.Event` object
             Light class encapsulating signal, onset, pick and location
             information for a given event.
 
         Returns
         -------
-        p_pick : `obspy.UTCDateTime` object
+        p_pick : `obspy.UTCDateTime` object or "-1"
             P pick time. Autopick time if available, otherwise ModelledTime
             calculated by the autopicker.
-        s_pick : `obspy.UTCDateTime` object
+        s_pick : `obspy.UTCDateTime` object or "-1"
             S pick time. Autopick time if available, otherwise ModelledTime
             calculated by the autpicker.
-        picked : bool
-            Whether at least one of the phases was picked by the autopicker.
 
         """
 
         picks = event.picks["df"]
-        p_pick = picks.iloc[i*2]["PickTime"]
-        s_pick = picks.iloc[i*2+1]["PickTime"]
-
-        picked = True
-        # If neither P nor S is picked, picked=False
-        if p_pick == -1 and s_pick == -1:
-            picked = False
-        if p_pick == -1:
-            p_pick = picks.iloc[i*2]["ModelledTime"]
-        if s_pick == -1:
-            s_pick = picks.iloc[i*2+1]["ModelledTime"]
+        picks = picks.loc[picks["Station"] == station]
 
-        # Convert to UTCDateTime objects
-        p_pick = UTCDateTime(p_pick)
-        s_pick = UTCDateTime(s_pick)
+        if len(picks) > 0:
+            try:
+                p_pick = picks.loc[picks["Phase"] == "P"]["PickTime"].iloc[0]
+                p_pick = UTCDateTime(str(p_pick))
```

Comment:
Out of curiousity, is the cast to `str` necessary?

Reply:
Possibly something to do with forcing a failure when the pick = -1?

Reply:
Yep. If it reads -1 from the picks dataframe it is an int, which gives a date in 1969 in UTCDateTime. I thought this way gave an easier to read catch (UTCDateTime("-1") --> ValueError --> check isinstance(UTCDateTime))